tests/components/blebox/conftest.py

```diff
@@ -0,0 +1,147 @@
+    return MockConfigEntry(
+        domain=const.DOMAIN, data={CONF_HOST: ip_address, CONF_PORT: 80},
+    )
+
+
+class DefaultBoxTest:
+    """Base class with methods common to BleBox integration tests."""
+
+    IP = "172.0.0.1"
+    HASS_TYPE = None  # Must be set in subclass
+
+    async def async_mock_entities(self, hass):
+        """Return a new entities configured through HASS."""
+
+        config = mock_config(self.IP)
+        config.add_to_hass(hass)
+
+        mod = self.HASS_TYPE
+
+        all_entries = []
+
+        def add_entries(entries, update):
+            for entry in entries:
+                entry.hass = hass
+                all_entries.append(entry)
+
+        assert await mod.async_setup_entry(hass, config, add_entries) is True
+        return all_entries
+
+    async def async_updated_entity(self, hass, index):
+        """Return an already-updated entity created through HASS."""
+        entity = (await self.async_mock_entities(hass))[index]
+        await entity.async_update()
+        return entity
+
+    async def test_setup_failure(self, hass):
+        """Test that setup failure is handled."""
+
+        path = "homeassistant.components.blebox.Products"
+        patcher = patch(path, mock.DEFAULT, blebox_uniapi.products.Products, True, True)
+        products_class = patcher.start()
+
+        products_class.async_from_host = CoroutineMock(
+            side_effect=blebox_uniapi.error.ClientError
+        )
+
+        with patch("homeassistant.components.blebox._LOGGER.error") as error:
+            with pytest.raises(PlatformNotReady):
+                await self.async_mock_entities(hass)
+
+            error.assert_has_calls(
+                [call("Identify failed at %s:%d (%s)", "172.0.0.1", 80, mock.ANY,)]
+            )
+            assert isinstance(error.call_args[0][3], blebox_uniapi.error.ClientError)
+
+    async def test_setup_failure_on_connection(self, hass):
+        """Test that setup failure is handled."""
+
+        path = "homeassistant.components.blebox.Products"
+        patcher = patch(path, mock.DEFAULT, blebox_uniapi.products.Products, True, True)
+        products_class = patcher.start()
+
+        products_class.async_from_host = CoroutineMock(
+            side_effect=blebox_uniapi.error.ConnectionError
+        )
+
+        with patch("homeassistant.components.blebox._LOGGER.error") as error:
+            with pytest.raises(PlatformNotReady):
+                await self.async_mock_entities(hass)
+
+            error.assert_has_calls(
+                [call("Identify failed at %s:%d (%s)", "172.0.0.1", 80, mock.ANY,)]
+            )
+            assert isinstance(
+                error.call_args[0][3], blebox_uniapi.error.ConnectionError
+            )
+
+    def default_mock(self):
+        """Return a default entity mock."""
+        raise NotImplementedError("Implement me")  # pragma: no cover
+
+    async def test_update_failure(self, hass):
```

Comment:
New tests need to be standalone pytest test functions.  If the function starts with `test_`, please move it out of the class